libs/designer/src/lib/core/queries/connector.ts

```diff
@@ -120,13 +120,8 @@ export const getLegacyDynamicSchema = async (
     return null;
   }
 
-  const schemaPath = extension['value-path'] ? extension['value-path'].split('/') : undefined;
-  return schemaPath
-    ? getObjectPropertyValue(
-        response,
-        schemaPath.length && equals(schemaPath[schemaPath.length - 1], 'properties') ? schemaPath.splice(-1, 1) : schemaPath
-      ) ?? null
-    : { properties: response, type: Types.Object };
+  const schemaPath = extension['value-path'] ? extension['value-path'].split('/').filter((s) => s) : undefined;
```

Comment:
NIT: Should the filter be more specific? e.g. its null or undefined or !isEmptyString

Reply:
its a split on string, so it will always be empty string not a number or object or array, hence just used simple notation for non-empty